File: baekjoon/25304.go
package main

import "fmt"

func main() {
	var x, n, a, b, sum int
	var res string

	fmt.Scanf("%d", &x)
	fmt.Scanf("%d", &n)

	for i := 0; i < n; i++ {
		fmt.Scanf("%d %d", &a, &b)
		sum += a * b
	}

	if sum == x {
		res = "Yes"
	} else {
		res = "No"
	}

	fmt.Printf("%s\n", res)
}

File: baekjoon/25304.py
sum = int(input())
n = int(input())

for _ in range(n):
    a, b = input().split()
    a = int(a)
    b = int(b)
    sum = sum - (a*b)

if sum == 0:
    print("Yes")
else:
    print("No")
